python/python/psyche/vllm/rust_bridge.py

```diff
@@ -0,0 +1,137 @@
+import logging
+from typing import Dict, Any, Optional
+from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+# Global registry of engines by ID
+_engines: Dict[str, Any] = {}
+
+
+def create_engine(
+    engine_id: str,
+    model_name: str,
+    tensor_parallel_size: int = 1,
+    dtype: str = "auto",
+    max_model_len: Optional[int] = None,
+    gpu_memory_utilization: float = 0.90,
+) -> Dict[str, Any]:
+    try:
+        from psyche.vllm.engine import UpdatableLLMEngine
+
+        logger.info(f"Creating engine '{engine_id}' with model '{model_name}'")
+
+        engine = UpdatableLLMEngine(
+            model_name=model_name,
+            tensor_parallel_size=tensor_parallel_size,
+            dtype=dtype,
+            max_model_len=max_model_len,
+            gpu_memory_utilization=gpu_memory_utilization,
+        )
+
+        _engines[engine_id] = engine
+
+        logger.info(f"Engine '{engine_id}' created successfully")
+
+        return {"status": "success", "engine_id": engine_id}
+
+    except Exception as e:
+        error_msg = f"Failed to create engine '{engine_id}': {e}"
+        logger.error(error_msg)
+        return {"status": "error", "error": error_msg}
+
+
+def run_inference(
+    engine_id: str,
+    prompt: str,
+    temperature: float = 1.0,
+    top_p: float = 1.0,
+    max_tokens: int = 100,
+) -> Dict[str, Any]:
+    try:
+        if engine_id not in _engines:
+            error_msg = f"Engine '{engine_id}' not found"
+            logger.error(error_msg)
+            return {"status": "error", "error": error_msg}
+
+        engine = _engines[engine_id]
```

Comment:
We're not checking if the `engine_id` is already present in the map so we could be overriding it here, maybe we should check before adding it.